bin/sessionmirror.py

```diff
@@ -0,0 +1,749 @@
+        # Write xml to file
+        if not op.exists(assr_cache_dir):
+            os.makedirs(assr_cache_dir)
+        assr_xml = src_assr.get()
+        xml_path = op.join(assr_cache_dir, 'assr.xml')
+        write_xml(assr_xml, xml_path)
+        dst_assr.create(xml=xml_path, allowDataDeletion=False)
+
+    # Process each resource of assr
+    for src_res in src_assr.out_resources():
+        res_label = src_res.label()
+        print('INFO:Processing resource:%s...' % res_label)
+        dst_res = dst_assr.out_resource(res_label)
+        res_cache_dir = op.join(assr_cache_dir, res_label)
+
+        if res_label == 'SNAPSHOTS':
+            copy_res(src_res, dst_res, res_cache_dir)
+        else:
+            copy_res(src_res, dst_res, res_cache_dir, use_zip=True)
+
+def write_xml(xml_str, file_path, clean_tags=True):
+    """Writing XML."""
+    root = ET.fromstring(xml_str)
+
+    # We only want the tags and attributes relevant to root, no children
+    if clean_tags:
+        # Remove ID
+        if 'ID' in root.attrib:
+            del root.attrib['ID']
+
+        # Remove sharing tags
+        tag = '{http://nrg.wustl.edu/xnat}sharing'
+        for child in root.findall(tag):
+            root.remove(child)
+
+        # Remove out
+        for child in root.findall('{http://nrg.wustl.edu/xnat}out'):
+            root.remove(child)
+            break
+
+        # Remove session ID
+        tag = '{http://nrg.wustl.edu/xnat}imageSession_ID'
+        for child in root.findall(tag):
+            root.remove(child)
+
+        # Remove subject ID
+        for child in root.findall('{http://nrg.wustl.edu/xnat}subject_ID'):
+            root.remove(child)
+
+        # Remove _session ID
+        tag = '{http://nrg.wustl.edu/xnat}image_session_ID'
+        for child in root.findall(tag):
+            root.remove(child)
+
+        # Remove scans
+        for child in root.findall('{http://nrg.wustl.edu/xnat}scans'):
+            root.remove(child)
+            break
+
+        # Remove assessors
+        for child in root.findall('{http://nrg.wustl.edu/xnat}assessors'):
+            root.remove(child)
+            break
+
+        # Remove resources
+        for child in root.findall('{http://nrg.wustl.edu/xnat}resources'):
+            root.remove(child)
+            break
+
+        # Remove experiments
+        for child in root.findall('{http://nrg.wustl.edu/xnat}experiments'):
+            root.remove(child)
+            break
+
+    try:
+        # Write to file
+        ET.register_namespace('xnat', 'http://nrg.wustl.edu/xnat')
+        ET.register_namespace('proc', 'http://nrg.wustl.edu/proc')
+        ET.register_namespace('prov', 'http://www.nbirn.net/prov')
+        ET.register_namespace('fs', 'http://nrg.wustl.edu/fs')
+        ET.ElementTree(root).write(file_path)
+    except IOError as error:
+        print('ERROR:writing xml file: {}: {}'.format(file_path, str(error)))
+
+
+def check_attributes(src_obj, dest_obj, dtype=None):
```

Comment:
function `check_attributes` is defined but seems to never be used

Reply:
This part was hard to follow in the original code (mostly commented out).
(see https://github.com/VUIIS/dax/blob/master/bin/Xnat_tools/Xnatmirror#L461)

Again we may just remove the function if not used.

Reply:
I see, is a sort of additional control mechanism for checking the consistency of the resource attributes. We could exclude it easing the maintenance of ported code but keeping the function as is now is not hurting anyway.